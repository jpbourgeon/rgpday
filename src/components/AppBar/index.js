import React from 'react'
import PropTypes from 'prop-types'
import { withStyles } from '@material-ui/core/styles'
import AppBar from '@material-ui/core/AppBar'
import { Link } from '@reach/router'
import Toolbar from '@material-ui/core/Toolbar'
import Typography from '@material-ui/core/Typography'
import Hidden from '@material-ui/core/Hidden'
import Info from '@material-ui/icons/Info'
import Email from '@material-ui/icons/Email'
import AccountCircle from '@material-ui/icons/AccountCircle'

const styles = theme => ({
  root: {
    paddingBottom: theme.spacing.unit * 4
  },
  grow: {
    flexGrow: 1
  },
  menuLink: {
    fontFamily: 'Roboto',
    fontSize: 'bold',
    color: theme.palette.common.white,
    textDecoration: 'none',
    margin: '0 0.5em'
  }
})

const AppBarComponent = props => {
  const { classes } = props
  return (
    <div className={classes.root}>
      <AppBar position='static'>
        <Toolbar>
          <Typography variant='h6' color='inherit' className={classes.grow}>
            <Link to='/' className={classes.menuLink}>RGPDay</Link>
          </Typography>
          <Link to='/about' className={classes.menuLink}>
            <Hidden smUp><Info /></Hidden>
            <Hidden xsDown>Ã€ propos</Hidden>

          </Link>
          <Link to='/contact' className={classes.menuLink}>
            <Hidden smUp><Email /></Hidden>
            <Hidden xsDown>Contact</Hidden>
          </Link>
          <Link to='/login' className={classes.menuLink}>
            <Hidden smUp><AccountCircle /></Hidden>
            <Hidden xsDown>Login</Hidden>
          </Link>
        </Toolbar>
      </AppBar>
    </div>
  )
}

AppBarComponent.propTypes = {
  classes: PropTypes.object.isRequired
}

export default withStyles(styles)(AppBarComponent)
