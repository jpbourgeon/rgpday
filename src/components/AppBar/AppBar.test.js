/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import AppBar from './'
import jssSerializer from '../../../utils/jssSnapshotSerializer'

expect.addSnapshotSerializer(jssSerializer)

describe('The AppBar component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<AppBar />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
