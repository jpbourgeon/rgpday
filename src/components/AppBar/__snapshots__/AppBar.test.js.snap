// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The AppBar component should render correctly 1`] = `
<div
  className="root-1"
>
  <header
    className="root-13 elevation4-19 root-4 positionStatic-8 colorPrimary-11"
  >
    <div
      className="root-40 regular-42 gutters-41"
    >
      <h6
        className="root-44 h6-61 colorInherit-73 grow-2"
      >
        <a
          aria-current="page"
          className="menuLink-3"
          href="/"
          onClick={[Function]}
        >
          RGPDay
        </a>
      </h6>
      <a
        className="menuLink-3"
        href="/about"
        onClick={[Function]}
      >
        Ã€ propos
      </a>
      <a
        className="menuLink-3"
        href="/contact"
        onClick={[Function]}
      >
        Contact
      </a>
      <a
        className="menuLink-3"
        href="/login"
        onClick={[Function]}
      >
        Login
      </a>
    </div>
  </header>
</div>
`;
